dev/integration_tests/flutter_gallery/lib/demo/shrine/model/app_state_model.dart

```diff
@@ -70,16 +70,14 @@ class AppStateModel extends Model {
 
   // Removes an item from the cart.
   void removeItemFromCart(int productId) {
-    final int? value = _productsInCart[productId];
-
-    if (value != null) {
-      if (_productsInCart[productId] == 1) {
+    switch (_productsInCart[productId]) {
+      case null:
+        return;
```

Comment:
Previously, `notifyListeners` was called in this case, and now it won't be. Is that an intentional change?

Reply:
Yes, I noticed that previously there was a notification being sent even when the ChangeNotifier hadn't made any changes.

There's an interesting section of the [`setState` docs](https://main-api.flutter.dev/flutter/widgets/State/setState.html#design-discussion) explaining that they moved away from `markNeedsBuild` for this exact reason.

My preference would be to leave this change here, though we could revert it in order to make this a "pure refactor".

Reply:
I would personally lean towards keeping this as a pure refactor (otherwise it will be much harder to land).